### ISSUE TITLE
Fine_Tunning with isnet-general-use.pth

### ISSUE THREAD

**Issue by kabbas570** [xuebinqin/DIS#105]
Hello, thanks for providing the code publicly.
I have a question about fine-tuning the network, can I start training with **isnet-general-use.pth** weigh file and further fine-tune these weights i.e. transfer learning?

Cheers
Abbas

**Comment by kabbas570** [comment]
I found this code from train_valid_inference_main.py

```
if(hypar["gt_encoder_model"]!=""):
        model_path = hypar["model_path"]+"/"+hypar["gt_encoder_model"]
        if torch.cuda.is_available():
            net.load_state_dict(torch.load(model_path))
            net.cuda()
        else:
            net.load_state_dict(torch.load(model_path,map_location="cpu"))
        print("gt encoder restored from the saved weights ...")
        return net ############
```

Does it mean it will load the weights for both the encoder and decoder of the model .pth file specified at hypar["model_path"] if we specify hypar["gt_encoder_model"] = ' isnet-general-use.pth'?

**Comment by youzipi** [comment]
maybe the `restore_model`?
```py
hypar["restore_model"] = "RMBG-1.4.pth" ## name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
```
@kabbas570

**Comment by hjj-lmx** [comment]
> ```python
> name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
> ```
Excuse me, has the problem been resolved?

**Comment by hjj-lmx** [comment]
> gt_encoder_model
May I ask how to continue training on isnet general use. pth? Have you resolved this?

**Comment by youzipi** [comment]
> > ```python
> > name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
> > ```
> 
> Excuse me, has the problem been resolved?

this parameter works.

**Comment by hjj-lmx** [comment]
> > > ```python
> > > name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
> > > ```
> > 
> > 
> > 请问，问题解决了吗？
> 
> 这个参数有效。

![1715829207(1)](https://github.com/xuebinqin/DIS/assets/107909464/2da1c3a5-5ba6-41dd-bd03-68a1ebbc4087)
This is my input parameter. I have received 200 training sets, and only one model file will be generated during training, which is an improvement compared to isnet general use. pth, but the effect is not very good. What is the reason for this situation

**Comment by youzipi** [comment]
> > > > ```python
> > > > name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
> > > > ```
> > > 
> > > 
> > > 请问，问题解决了吗？
> > 
> > 
> > 这个参数有效。
> 
> ![1715829207(1)](https://private-user-images.githubusercontent.com/107909464/331036869-2da1c3a5-5ba6-41dd-bd03-68a1ebbc4087.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4NDgzNTgsIm5iZiI6MTcxNTg0ODA1OCwicGF0aCI6Ii8xMDc5MDk0NjQvMzMxMDM2ODY5LTJkYTFjM2E1LTViYTYtNDFkZC1iZDAzLTY4YTFlYmJjNDA4Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNlQwODI3MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTc5YzBkZTgxYjYzYzgzNTE3MTVjODI0MTk2MmYzNGVlYThjOThkNDJjODIyODJjNDdkZWVjMDFlODNiNjkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NwB82JZ840FUfUGNGrbPjgFjm3UDhQqSht4ARa2nen0) This is my input parameter. I have received 200 training sets, and only one model file will be generated during training, which is an improvement compared to isnet general use. pth, but the effect is not very good. What is the reason for this situation

is the loss convergence ?
do you use the best f-score snapshot?

if yes, you need to expand your train set.
or if your case is similar to the origin model, you can check the `interm_sup` parameter, it freeze the origin parameter.

**Comment by youzipi** [comment]
> > > > > > ```python
> > > > > > name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
> > > > > > ```
> > > > > 
> > > > > 
> > > > > 请问，问题解决了吗？
> > > > 
> > > > 
> > > > 这个参数有效。
> > > 
> > > 
> > > ![1715829207(1)](https://private-user-images.githubusercontent.com/107909464/331036869-2da1c3a5-5ba6-41dd-bd03-68a1ebbc4087.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4NDgzNTgsIm5iZiI6MTcxNTg0ODA1OCwicGF0aCI6Ii8xMDc5MDk0NjQvMzMxMDM2ODY5LTJkYTFjM2E1LTViYTYtNDFkZC1iZDAzLTY4YTFlYmJjNDA4Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNlQwODI3MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTc5YzBkZTgxYjYzYzgzNTE3MTVjODI0MTk2MmYzNGVlYThjOThkNDJjODIyODJjNDdkZWVjMDFlODNiNjkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NwB82JZ840FUfUGNGrbPjgFjm3UDhQqSht4ARa2nen0) This is my input parameter. I have received 200 training sets, and only one model file will be generated during training, which is an improvement compared to isnet general use. pth, but the effect is not very good. What is the reason for this situation
> > 
> > 
> > is the loss convergence ? do you use the best f-score snapshot?
> > if yes, you need to expand your train set. or if your case is similar to the origin model, you can check the `interm_sup` parameter, it freeze the origin parameter.
> 
> interm_sup = false We are using the default parameters provided on Git, only changing the dataset passed in

你加我 v 吧，eW91emlwaXBwaQ==

**Comment by hjj-lmx** [comment]
> eW91emlwaXBwaQ==

What kind of account is this

**Comment by hjj-lmx** [comment]
> > > > > > > ```python
> > > > > > > name of the segmentation model weights .pth for resume training process from last stop or for the inferencing
> > > > > > > ```
> > > > > > 
> > > > > > 
> > > > > > 请问，问题解决了吗？
> > > > > 
> > > > > 
> > > > > 这个参数有效。
> > > > 
> > > > 
> > > > ![1715829207(1)](https://private-user-images.githubusercontent.com/107909464/331036869-2da1c3a5-5ba6-41dd-bd03-68a1ebbc4087.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4NDgzNTgsIm5iZiI6MTcxNTg0ODA1OCwicGF0aCI6Ii8xMDc5MDk0NjQvMzMxMDM2ODY5LTJkYTFjM2E1LTViYTYtNDFkZC1iZDAzLTY4YTFlYmJjNDA4Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNlQwODI3MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTc5YzBkZTgxYjYzYzgzNTE3MTVjODI0MTk2MmYzNGVlYThjOThkNDJjODIyODJjNDdkZWVjMDFlODNiNjkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NwB82JZ840FUfUGNGrbPjgFjm3UDhQqSht4ARa2nen0)这是我的输入参数。我收到了200个训练集，训练时只会生成一个模型文件，相对于isnet一般使用来说是一个进步。 pth，但是效果不是很好。造成这种情况的原因是什么
> > > 
> > > 
> > > 是损失收敛吗？你使用最好的 f 分数快照吗？
> > > 如果是，您需要扩展您的火车组。或者如果您的情况与原始模型类似，您可以检查参数`interm_sup`，它会冻结原始参数。
> > 
> > 
> > interm_sup = false 我们使用Git上提供的默认参数，仅更改传入的数据集
> 
> 你加我吧，eW91emlwaXBwaQ==

How to add this

**Comment by youzipi** [comment]
> > eW91emlwaXBwaQ==
> 
> What kind of account is this

微信，youzipippi

**Comment by xiefeihua** [comment]
Hello, I have the same issue. Is there any way to solve it? I am looking forward to your response.
